grpc/src/client/load_balancing/child_manager.rs

```diff
@@ -64,44 +71,48 @@ pub struct ChildUpdate<T> {
     /// The identifier the ChildManager should use for this child.
     pub child_identifier: T,
     /// The builder the ChildManager should use to create this child if it does
-    /// not exist.
+    /// not exist.  The child_policy_builder's name is effectively a part of the
+    /// child_identifier.  If two identifiers are identical but have different
+    /// builder names, they are treated as different children.
     pub child_policy_builder: Arc<dyn LbPolicyBuilder>,
-    /// The relevant ResolverUpdate to send to this child.
-    pub child_update: ResolverUpdate,
+    /// The relevant ResolverUpdate and LbConfig to send to this child.  If
+    /// None, then resolver_update will not be called on the child.  Should
+    /// generally be Some for any new children, otherwise they will not be
+    /// called.
+    pub child_update: Option<(ResolverUpdate, Option<LbConfig>)>,
 }
 
 pub trait ResolverUpdateSharder<T>: Send {
-    /// Performs the operation of sharding an aggregate ResolverUpdate into one
-    /// or more ChildUpdates.  Called automatically by the ChildManager when its
-    /// resolver_update method is called.  The key in the returned map is the
-    /// identifier the ChildManager should use for this child.
+    /// Performs the operation of sharding an aggregate ResolverUpdate/LbConfig
+    /// into one or more ChildUpdates.  Called automatically by the ChildManager
+    /// when its resolver_update method is called.
     fn shard_update(
-        &self,
+        &mut self,
         resolver_update: ResolverUpdate,
-    ) -> Result<Box<dyn Iterator<Item = ChildUpdate<T>>>, Box<dyn Error + Send + Sync>>;
+        update: Option<LbConfig>,
+    ) -> Result<impl Iterator<Item = ChildUpdate<T>>, Box<dyn Error + Send + Sync>>;
 }
 
-impl<T> ChildManager<T> {
+impl<T: Debug, S> ChildManager<T, S>
+where
+    S: ResolverUpdateSharder<T>,
+{
     /// Creates a new ChildManager LB policy.  shard_update is called whenever a
     /// resolver_update operation occurs.
-    pub fn new(
-        update_sharder: Box<dyn ResolverUpdateSharder<T>>,
-        runtime: Arc<dyn Runtime>,
-    ) -> Self {
+    pub fn new(update_sharder: S, runtime: Arc<dyn Runtime>) -> Self {
         Self {
             update_sharder,
             subchannel_child_map: Default::default(),
             children: Default::default(),
             pending_work: Default::default(),
             runtime,
+            updated: false,
         }
     }
 
     /// Returns data for all current children.
-    pub fn child_states(&mut self) -> impl Iterator<Item = (&T, &LbState)> {
-        self.children
-            .iter()
-            .map(|child| (&child.identifier, &child.state))
+    pub fn children(&mut self) -> impl Iterator<Item = &Child<T>> {
```

Comment:
Can this method accept an immutable reference instead?

Reply:
Yes, that should be fine.  I am not sure why I made these `&mut self` since they are read-only operations.  I'm expecting all real-world uses will have the ChildManager mutably when calling into it, but there's no reason to make these `&mut`.